### PR TITLE
[OAP-1647][oap-native-sql] row to columnar for decimal

### PR DESCRIPTION
**Pull request by rui-mo**
This PR adds support for decimal in the RowToColumnar.
 
fixed https://github.com/Intel-bigdata/OAP/issues/1647

### CODE REVIEW

**Comment by rui-mo** [comment]
@xuechendi could you try and review this patch? I can have correct result for q83 on my env with preferColumnar = true.

**Comment by xuechendi** [comment]
Looks great to me, I think we can merge this firstly. @zhouyuan

**Comment by xuechendi** [comment]
@rui-mo , @zhouyuan , verified locally, Q83 is now correct.

File: oap-native-sql/core/src/main/scala/com/intel/oap/expression/ColumnarUnaryOperator.scala [hunk 1]
```diff
@@ -169,9 +169,7 @@ class ColumnarCast(child: Expression, datatype: DataType, timeZoneId: Option[Str
           if float.getPrecision() == FloatingPointPrecision.DOUBLE => 21
         case date: ArrowType.Date if date.getUnit == DateUnit.DAY => 10
         case decimal : ArrowType.Decimal =>
-          val precision = decimal.getPrecision()
-          val scale  = decimal.getScale()
-          precision
+          (decimal.getPrecision() + 2)
```
Comment (zhouyuan) [inline oap-native-sql/core/src/main/scala/com/intel/oap/expression/ColumnarUnaryOperator.scala:172]: magic number 2?

Comment (rui-mo) [reply oap-native-sql/core/src/main/scala/com/intel/oap/expression/ColumnarUnaryOperator.scala:172]: no, the main change is this:  https://github.com/Intel-bigdata/OAP/pull/1649/files#diff-f17cfe4ae8804014360b8ea71a2041e8R195. We found the data is wrong after row to columnar.

**Pull request merged**